qiskit_nature/drivers/second_quantization/bosonic_molecule_driver.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+This module implements a common molecule bosonic based driver.
+"""
+
+from typing import Optional, Dict, Any
+import logging
+from enum import Enum
+
+from qiskit.exceptions import MissingOptionalLibraryError
+from .bosonic_driver import BosonicDriver
+from ..molecule import Molecule
+from .gaussiand import GaussianForcesDriver
+
+logger = logging.getLogger(__name__)
+
+
+class BosonicDriverType(Enum):
+    """Bosonic Driver Type."""
+
+    AUTO = "auto"
+    GAUSSIAN_FORCES = "gaussian_forces"
+
+    @staticmethod
+    def driver_class_from_type(driver_type: "BosonicDriverType"):
+        """Get driver class from driver type"""
```

Comment:
Missing `Args` and `Raises` documentation
Missing returned mypy type

Reply:
Added the missing docs and type